### ISSUE TITLE
Blurry images/resolution issue

### ISSUE THREAD

**Issue by ViktorKahu** [you-apps/WallYou#127]
### Steps to reproduce

Ever since the update on the 14th, the lockscreen images went extremely bad resolution, blurry.

I'm guessing it has something to do with "Improve wallpaper grid layout on large screens by @SuhasDissa"

### Expected behavior

As before the update.

### Actual behavior

![Screenshot_20230914-184302~3](https://github.com/you-apps/WallYou/assets/125934998/9f16908e-7b10-4929-b147-635e9c918986)

### Wall You version

6.0

### Android version

13 GrapheneOS

### Other details

Local images are used for lockscreen wallpaper.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by SuhasDissa** [comment]
I don't think the layout change is related to this problem at all.
I guess the newly added wallpaper sources are causing this issue.

Can you give us some more information ?
- What wallpaper source did you use ?
- Did the image look blurry on home screen as well ?

**Comment by ViktorKahu** [comment]
To answer your questions:

Image sources:

Lockscreen: phone (local folder).
Homescreen: favorites

The problem is on lockscreen only, but I have not tested configurations other than the one above.

I can assure you the images in my local folder are in high resolution and everything worked good with the same setup prior to the update. Clearing cache changed nothing.

**Comment by ViktorKahu** [comment]
After further testing, the problem seems to be specifically with using a locally stored collections, which should narrow down the fault finding.

I tried a different configuration - favorites to the lock screen and local folder to the homescreen. The problem moved in sympathy.

Most (but not all) of my images are 2268 x 4032 resolution.

Another difference I noticed is that the collections images are in .png format, where mine are .jpg.

It's something to do with the way local images are processed by the app, may be a cache issue or a scaling issue.

**Comment by KR4U7H3R** [comment]
I missed the last few updates due to signature change. Now I can confirm this problem. I also use local stored images in .jpg format.

**Comment by ViktorKahu** [comment]
Actually, the problem is deeper then what's described above so far.

It seems yhe local images mechanism is broken completely with the 6.0 update.

Observations:

1) what is described above is only relevant when upgrading from 5.0 to 6.0. This means that the local folder is set up and working in 5.0 and the images go blurry and terrible when upgraded to 6.0.

2) Clearing cache, app data, deleting and reinstalling the app and restarting the phone achieves nothing.

3) In 6.0, if you change the local images folder, it breaks completely and the wallpaper changer stops working. The wallpaper gets stuck on the last image that was used.

4) The same story goes for using any local folder for the first time in 6.0 if it wasn't already set up in 5.0.

5) Online images work with no issues.

I tested it on two different devices with the same results, one running A13 GrapheneOS, the other Android 7.1 Nougat 

I had to downgrade to 5.0 to get it working.
